rpc/events.go

```diff
@@ -100,6 +105,87 @@ func (h *Handler) SubscribeNewHeads(ctx context.Context) (uint64, *jsonrpc.Error
 	return id, nil
 }
 
+// SubscribeTxnStatus subscribes to status changes of a transaction. It checks for updates each time a new block is added.
+// Subsequent updates are sent only when the transaction status changes.
+// The optional block_id parameter is ignored, as status changes are not stored and historical data cannot be sent.
+func (h *Handler) SubscribeTxnStatus(ctx context.Context, txHash felt.Felt, _ *BlockID) (*SubscriptionID, *jsonrpc.Error) {
+	var (
+		lastKnownStatus, lastSendStatus *TransactionStatus
+		wrapResult                      = func(s *TransactionStatus) *NewTransactionStatus {
+			return &NewTransactionStatus{
+				TransactionHash: &txHash,
+				Status:          s,
+			}
+		}
+	)
+
+	w, ok := jsonrpc.ConnFromContext(ctx)
+	if !ok {
+		return nil, jsonrpc.Err(jsonrpc.MethodNotFound, nil)
+	}
+
+	id := h.idgen()
+	subscriptionCtx, subscriptionCtxCancel := context.WithCancel(ctx)
+	sub := &subscription{
+		cancel: subscriptionCtxCancel,
+		conn:   w,
+	}
+
+	lastKnownStatus, rpcErr := h.TransactionStatus(subscriptionCtx, txHash)
+	if rpcErr != nil {
+		h.log.Warnw("Failed to get Tx status", "txHash", &txHash, "rpcErr", rpcErr)
+		return nil, rpcErr
+	}
+
+	h.mu.Lock()
+	h.subscriptions[id] = sub
+	h.mu.Unlock()
+
+	statusSub := h.txnStatus.Subscribe()
+	headerSub := h.newHeads.Subscribe()
+	sub.wg.Go(func() {
+		defer func() {
+			h.unsubscribe(sub, id)
+			statusSub.Unsubscribe()
+			headerSub.Unsubscribe()
+		}()
+
+		if err := h.sendTxnStatus(sub.conn, wrapResult(lastKnownStatus), id); err != nil {
+			h.log.Warnw("Error while sending Txn status", "txHash", txHash)
```

Comment:
maybe correct severity would be `Error` instead of `Warnw` ? Also please add `"err", err` as last arguments